packages/flutter/lib/src/widgets/text_editing_intents.dart

```diff
@@ -0,0 +1,125 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'actions.dart';
+
+/// An [Intent] to expand the selection left to the start/end of the current
+/// line.
+///
+/// {@template flutter.widgets.TextEditingIntents.seeAlso}
+/// See also:
+///
+///   * [DefaultTextEditingActions], which responds to this [Intent].
+///   * [DefaultTextEditingShortcuts], which triggers this [Intent].
+/// {@endtemplate}
+class ExpandSelectionLeftByLineTextIntent extends Intent {}
```

Comment:
I was able to add const constructors to all of these Intents, great suggestion.  It didn't seem to work for TextEditingAction because ContextAction and Action have no const constructor.